units/systemd-resolv-conf.service

```diff
@@ -0,0 +1,27 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Create resolv.conf if not already exists
+Documentation=man:systemd-resolved.service(8) man:resolv.conf(5)
+After=systemd-journald.service
+After=local-fs.target network-pre.target systemd-sysusers.service
+; no dynamic DNS configuration would be known before network configuration
+After=systemd-networkd.service
+Before=network.target
+ConditionPathExists=/run/systemd
+ConditionPathExists=!/etc/resolv.conf
+BindsTo=systemd-resolved.service
+
+[Service]
+ExecStart=-/bin/ln -s ../run/systemd/resolve/stub-resolv.conf resolv.conf
```

Comment:
Because of the ConditionPathExists= above, this is only started when resolv.conf doesn't exist. So "-" is not appropriate, this should always succeed. And no need to write `/bin/ln`, just `ln` is fine.

Reply:
- Is required on Debian+Ubuntu, because there is race condition. It might have created the symlink after condition check and it would fail. Unit tests here discovered that.

Reply:
If there's a race condition, that is bad. What is the symlink created to if the race is won by the other side?

(Or in other words: designing a system so that it has a known race condition that results in divergent outcomes is not acceptable. Thing may happen in an unspecified order, but the end result must always be the same. If it is not, then we need to add additional ordering specifications.)